lib/streamlit/type_util.py

```diff
@@ -656,6 +657,24 @@ def ensure_indexable(obj: OptionSequence[V_co]) -> Sequence[V_co]:
         return list(it)
 
 
+def check_python_comparable(seq: Sequence[Any]) -> None:
+    """Check if the sequence elements support "python comparison".
+    That means that the equality operator (==) returns a boolean value.
+    Which is not True for e.g. numpy arrays and pandas series."""
+    try:
+        bool(seq[0] == seq[0])
+    except LookupError:
+        # In case of empty sequences, the check not raise an exception.
+        pass
+    except ValueError:
+        raise StreamlitAPIException(
+            "Options must be python comparable, that means to return a boolean for "
```

Comment:
I would phrase it more like
"Options must be comparable, returning a boolean when used with `==`.
Got {type(seq[0]).__name__}, which cannot be compared.
Refactor your code to use elements of comparable types as options,
e.g. use indices instead."

Broadly similar, but I think slightly clearer in grammar and from not introducing the phrase "python comparable" when it should be clear from context what "comparable" means.